src/components/PopupWithForm.js

```diff
@@ -3,33 +3,40 @@ import Popup from "./Popup.js";
 export default class PopupWithForm extends Popup {
   constructor(popupSelector, handleFormSubmit) {
     super({ popupSelector });
+
     this._popupForm = this._popupElement.querySelector(".modal__form");
-    this._formButton = this._popupElement.querySelector(".modal__button");
-    //this._formButtonText = this._formButton.textContent;//
+    this._submitButton = this._popupElement.querySelector(".modal__button");
+    this._submitButtonText = this._submitButton.textContent;
     this._handleFormSubmit = handleFormSubmit;
     this._inputList = this._popupElement.querySelectorAll(".modal__input");
   }
 
   _getInputValues() {
-    const inputValues = {};
-    this._inputList.forEach((input) => (inputValues[input.name] = input.value));
-    return inputValues;
-  }
-
-  setInputValues(data) {
+    const formValues = {};
     this._inputList.forEach((input) => {
-      input.value = data[input.name];
+      formValues[input.name] = input.value;
     });
+    return formValues;
   }
 
   setEventListeners() {
-    this._popupForm.addEventListener("submit", this._handleSubmit);
     super.setEventListeners();
+    this._popupForm.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this._handleFormSubmit(this._getInputValues());
+    });
   }
 
-  _handleSubmit = (evt) => {
-    evt.preventDefault();
-    this._handleFormSubmit(this._getInputValues());
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = this._submitButtonText;
+    }
+  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`